cmd/cstor-volume-mgmt/controller/volume-controller/handler.go

```diff
@@ -324,6 +404,71 @@ func (c *CStorVolumeController) getVolumeResource(
 	return cStorVolumeGot, nil
 }
 
+// resizeCStorVolume resize the cstorvolume and if any error occurs updates the
+// resize conditions of cstorvolume either with success or failure message
+func (c *CStorVolumeController) resizeCStorVolume(
+	cStorVolume *apis.CStorVolume) (*apis.CStorVolume, error) {
+	var eventMessage string
+	var err error
+	isResizeSuccess := false
+	cloneCV := cStorVolume.DeepCopy()
+	customCVObj := cstorvolume.NewForAPIObject(cloneCV)
+	// NOTE: We are processing resize process only after updating CV resize conditions
+	// so in below call there is no chance of geting new CVCondition instance
+	conditionStatus := customCVObj.GetCVCondition(apis.CStorVolumeResizing)
+	desiredCap := cloneCV.Spec.Capacity.String()
+
+	err = volume.ResizeTargetVolume(cloneCV)
+	if err != nil {
+		eventMessage = fmt.Sprintf(
+			"failed to resize cstorvolume %s with capacity %s error %v",
+			cloneCV.Name,
+			desiredCap,
+			err,
+		)
+		c.recorder.Event(cloneCV, corev1.EventTypeWarning, string(common.FailureUpdate), eventMessage)
+		conditionStatus.Message = eventMessage
+	} else {
+		// In success case remove resize condition
+		conditionStatus = apis.CStorVolumeCondition{}
+		cloneCV.Status.Capacity = cloneCV.Spec.Capacity
+		isResizeSuccess = true
+	}
+
+	newCV, cvUpdateErr := c.updateCVSubResource(cloneCV, conditionStatus)
+	if cvUpdateErr == nil && isResizeSuccess {
+		eventMessage = fmt.Sprintf(
+			"successfully resized volume %s to %s",
+			cloneCV.Name,
+			desiredCap,
+		)
+		c.recorder.Event(cloneCV, corev1.EventTypeNormal, string(common.SuccessUpdated), eventMessage)
+	}
+	return newCV, cvUpdateErr
+}
+
+func (c *CStorVolumeController) updateCVSubResource(
+	cstorVolume *apis.CStorVolume,
+	conditionStatus apis.CStorVolumeCondition) (*apis.CStorVolume, error) {
+	//cvCopy := cStorVolume.DeepCopy()
+	cvBuilder := cstorvolume.
+		BuilderFromAPI(cstorVolume).
+		WithCondition(conditionStatus)
+	cvObj, err := cvBuilder.Build()
```

Comment:
nit: build() can be used in same line i.e.
```go
	cvObj, err := cstorvolume.
		BuilderFromAPI(cstorVolume).
		WithCondition(conditionStatus).
                Build()
```

Reply:
Done